### ISSUE TITLE
Wrong prop types displayed in Storybook props table

### ISSUE THREAD

**Issue by mimarz** [digdir/designsystemet#2760]
When declaring a default value for native props in our preview story, wrong type is displayed in the props table. It uses the type for the value declared in the preview story instead of the actual type on the component. 

Notes;
- Alert has defined a `string` default value for [children](https://github.com/digdir/designsystemet/blob/f3041150919e042935c89de767bd79ef7c6d82ae/packages/react/src/components/Alert/Alert.stories.tsx#L24) in its preview story, resulting in `string` being shown as the prop type instead of the native one, `ReactNode`
- Maybe there is a solution in tweaking our [reactDocgen settings/parsing](https://github.com/digdir/designsystemet/blob/f3041150919e042935c89de767bd79ef7c6d82ae/apps/storybook/.storybook/main.ts#L11)

**Comment by elaffen** [comment]
Why should you list the children prop unless it's something else than the default ReactNode?
-	One reason could be that you want to add some valuable description/doc for the usage of children in the component.
-	You would like the type to be something else. For example 'string'
There is an option for reactDocgenTypescriptOptions which will let you skip the listing of children: skipChildrenPropWithoutDoc: false,

**Comment by mimarz** [comment]
Aye, having tested a bit this looks to be related specifically to the `children` prop. 

Its a good point of declaring jsdoc for `children` for more composed component such as `Accordion`, where we expect the `children` to be a multiple `Accordion.Item`, but for something like `Button` maybe not so? 

I agree we should document when `children` is expected to be a set of sub-components of a specific.

Lets do some testing with the `skipChildrenPropWithoutDoc` option. Maybe that can be a quick fix for now and we'll get some proper jsdoc for the `children` prop sorted later.

**Comment by elaffen** [comment]
Your propFilter prevents the children from being Docgenerated. That is why you need to define the prop yourself or tweak the filter. https://github.com/digdir/designsystemet/blob/6c59fc5652c8c78ef4b0b94faea05f2618989dab/apps/storybook/.storybook/main.ts#L19

But because you have specified that children arg in the Alert-story, Storybook guesses that children is of type 'string'
https://github.com/digdir/designsystemet/blob/6c59fc5652c8c78ef4b0b94faea05f2618989dab/packages/react/src/components/Alert/Alert.stories.tsx#L24

If you change the value of children in the story to true `children: true,` in Alert.stories.tsx line 24, Storybook will list the prop as boolean.

**Comment by mimarz** [comment]
> Your propFilter prevents the children from being Docgenerated. That is why you need to define the prop yourself or tweak the filter.
> 
> https://github.com/digdir/designsystemet/blob/6c59fc5652c8c78ef4b0b94faea05f2618989dab/apps/storybook/.storybook/main.ts#L19
> 
> But because you have specified that children arg in the Alert-story, Storybook guesses that children is of type 'string'
> 
> https://github.com/digdir/designsystemet/blob/6c59fc5652c8c78ef4b0b94faea05f2618989dab/packages/react/src/components/Alert/Alert.stories.tsx#L24
> 
> If you change the value of children in the story to true `children: true,` in Alert.stories.tsx line 24, Storybook will list the prop as boolean.

yeah, good catch! We added that filter because of performance problems locally with react-docgen scanning for types _everywhere_. We'll have to fix this somehow so that we get the correct types for children on all our components :)

**Comment by unekinn** [comment]
> We added that filter because of performance problems locally

Our `propFilter` is identical to Storybook's default `propFilter`<sup>1</sup>. Without this filter prop table in Storybook will also display every single prop that React allows for the component's element type in the props table, which makes the actually relevant documentation almost impossible to find.

I'm not sure if it's a good idea, but `propFilter` has a `(prop, component) => boolean` signature which allows us to look up the `component.name`. If we want to omit or include React props from the docs on a per-component basis, this could maybe be used.

<sup>1</sup>: ...except that we also omit `popovertarget`, which we have defined locally as a patch on React's types

**Comment by eirikbacker** [comment]
Is this relevant anymore, after docs-rewrite @Barsnes ? ☺️

**Comment by Barsnes** [comment]
> Is this relevant anymore, after docs-rewrite [@Barsnes](https://github.com/Barsnes) ? ☺️

hm it kinda is, but I'll close this since storybook is not so much public facing anymore. Our proptables should have all, and proper types
